dev/com.ibm.ws.repository.generator/src/com/ibm/ws/repository/generator/SingleJsonRepositoryGenerator.java

```diff
@@ -0,0 +1,254 @@
+        try {
+            assetTypeEnumValue = ResourceType.valueOf(assetType);
+        } catch (IllegalArgumentException iae) {
+            iae.printStackTrace();
+            throw new BuildException(iae);
+        }
+        
+        SingleFileRepositoryConnection mySingleFileRepo = null;
+        if (jsonFile.exists()) {
+        	mySingleFileRepo = new SingleFileRepositoryConnection(jsonFile);
+        } else {
+        	try {
+				mySingleFileRepo = SingleFileRepositoryConnection.createEmptyRepository(jsonFile);
+			} catch (IOException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+        } 
+        
+       // OutputStream os = null;
+        Parser<? extends RepositoryResourceWritable> parser = null;
+
+        try {
+            // PRODUCTSAMPLE,OPENSOURCE,INSTALL,ADDON,FEATURE,IFIX,ADMINSCRIPT,CONFIGSNIPPET,TOOL
+            switch (assetTypeEnumValue) {
+
+                case PRODUCTSAMPLE:
+                case OPENSOURCE:
+                    parser = new SamplesParser();
+                    break;
+
+                case CONFIGSNIPPET:
+//		      NOT YET IMPLEMENTED
+//                    parser = new ConfigSnippetParser(null);
+                    break;
+
+                case ADMINSCRIPT:
+//		      NOT YET IMPLEMENTED
+//                   parser = new AdminScriptParser(null);
+                    break;
+
+                case TOOL:
+                    parser = new ToolParser();
+                    break;
+
+                case ADDON:
+                case INSTALL:
+                    if (isPax(assetFile)) {
+                        parser = new PaxParser();
+                    } else if (isZip(assetFile)) {
+                        parser = new ProductZipParser();
+                    } else {
+                        parser = new ProductParser();
+                    }
+                    break;
+
+                case FEATURE:
+                    parser = new EsaParser();
+                    break;
+
+                case IFIX:
+//		      NOT YET IMPLEMENTED
+//                    parser = new IfixParser(null);
+                    break;
+
+                default:
+                    // we won't produce JSON for unlisted types
+            }
+            if (parser != null) {
+                RepositoryResourceWritable resource = parser.parseFileToResource(assetFile, metadataFile, null);
+                resource.updateGeneratedFields(true);
+                //os = getOutputStreamFromPath(getOutputLocation(), assetFile);
+                resource.setRepositoryConnection(mySingleFileRepo);
+                resource.uploadToMassive(new AddThenDeleteStrategy());
+                //resource.writeDiskRepoJSONToStream(os);
+            } else {
+                // Not a file we build json for
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new BuildException(e);
+        }
+        
+//        finally {
```

Comment:
Should this be removed?  (As well as the other commented out sections in this file)